BlueprintUI/Sources/Environment/EnvironmentKey.swift

```diff
@@ -26,4 +26,28 @@ public protocol EnvironmentKey {
     /// The default value that will be vended by an `Environment` for this key if no other value
     /// has been set.
     static var defaultValue: Self.Value { get }
+
+    /// Equivalency check on the `Value`s of `EnvironmentKey`s. This should return false if the
+    /// difference in values affects the measurement or layout of consuming elements.
+    static func isEquivalent(_ lhs: Value, _ rhs: Value) -> Bool
```

Comment:
Thought: Should we name these functions so that they describe that they're comparing only for layout/measurement purposes and not for strict content equability? `isEquivalentForLayout(...)`? I suppose not as I can't think of a good name...

Reply:
I suppose this question might be more relevant in [the next PR](https://github.com/square/Blueprint/pull/383/files#diff-a8aa53da00aac8da1ca838dad2feccf6c13b44e7e1d336db033af8e76aeab8ecR29-R32) (for the `ComparableElement` function)